src/core/resource.cc

```diff
@@ -650,6 +675,9 @@ resources allocate(configuration& c) {
     for (auto devid : c.devices) {
         ret.ioq_topology.emplace(devid, allocate_io_queues(topology, ret.cpus, cpu_to_node, c.num_io_groups, last_node_idx));
     }
+
+    ret.numa_node_id_to_cpuset = numa_node_id_to_cpuset(topology);
```

Comment:
The `std::unordered_map<unsigned, hwloc_obj_t> cpu_to_node` in `resource::allocate()` already contains `cpu -> numa_node` map, so the `numa_node_id_to_cpuset` can be evaluated by turning the `cpu_to_node` inside out. Is the newly introduced `numa_node_id_to_cpuset()` produces some different result?

Reply:
They aren't the same. For example, with --smp 1 cpu_to_node will be a map (cpu)0 -> (node)0, amd numa_node_id_to_cpuset() will return a map (node)0 -> (cpu)0..15. (node)1 -> (cpu)16..31 (say).

If we use the first for the prefault thread, it will bind the prefault thread to cpu 0 (worst place since it's a reactor thread), while with the second we bind to 0..15.